tests/pytorch/distributed/run_layer_with_overlap.py

```diff
@@ -87,6 +94,13 @@ def _get_layer_args(config, tp_group, tp_size, reference=False):
         kwargs["ub_bulk_dgrad"] = not config.overlap_rs_dgrad and not reference
         kwargs["ub_bulk_wgrad"] = not config.overlap_rs_dgrad and not reference
 
+    if config.ub_cfg is not None and isinstance(config.ub_cfg, str):
+        try:
+            import yaml
+        except ImportError:
+            raise RuntimeError("'yaml' package not found!")
```

Comment:
This is hacky.
```suggestion
```
Much better to add PyYAML as a test-time dependency: https://github.com/NVIDIA/TransformerEngine/blob/f3a009da7b577655fe3564982d3990b06aca53b1/setup.py#L109

Reply:
Addressed comment.